content/ko/docs/concepts/workloads/controllers/daemonset.md

```diff
@@ -0,0 +1,238 @@
+셀렉터의 기본 값은 `kubectl apply` 과 호환되지 않는다.
+또한, 한번 데몬셋이 만들어지면 `.spec.selector` 의 변형은 가능하지 않다.
+파드 셀렉터를 변형하면 의도하지 않게 파드는 고아가 되거나 사용자에게 혼란을 주는 것으로 밝혀졌다.
+
+`.spec.selector` 는 다음 2개의 필드로 구성된 오브젝트이다.
+
+* `matchLabels` - [레플리케이션 컨트롤러](/ko/docs/concepts/workloads/controllers/replicationcontroller/)의 `.spec.selector` 와 동일하게 작동한다.
+* `matchExpressions` - 키, 값 목록 그리고 키와 값과 관련된 연산자를
+  명시해서 보다 정교한 셀렉터를 만들 수 있다.
+
+2개가 명시되면 결과는 AND가 된다.
+
+만약 `.spec.selector` 를 명시하면, 이것은 `.spec.template.metadata.labels` 와 일치해야 한다. 일치하지 않는 구성은 API에 의해 거부된다.
+
+또한 일반적으로 다른 데몬셋이나 레플리카셋과 같은 다른 컨트롤러를 통해 직접적으로 
+레이블이 셀렉터와 일치하는 다른 파드를 생성하지 않아야 한다. 그렇지 않으면 데몬셋 
+컨트롤러는 해당 파드가 생성된 것으로 생각한다. 쿠버네티스는 이런 일을 하는 것을 
+막지 못한다. 이를 수행하고자 한다면 한 케이스로써 테스트할 노드에서 다른 값을 가지는 파드를 
+수동으로 생성하는 것이다.
+
+### 오직 일부 노드에서만 파드 실행
+
+만약 `.spec.template.spec.nodeSelector` 를 명시하면 데몬셋 컨트롤러는 
+[노드 셀렉터](/docs/concepts/configuration/assign-pod-node/#nodeselector)와 
+일치하는 노드에 파드를 생성한다. 마찬가지로 `.spec.template.spec.affinity` 를 명시하면 
+데몬셋 컨트롤러는 [노트 어피니티](/docs/concepts/configuration/assign-pod-node/#node-affinity)와 일치하는 노드에 파드를 생성한다. 
+만약 둘 중 하나를 명시하지 않으면 데몬셋 컨트롤러는 모든 노드에서 파드를 생성한다.
+
+## 데몬 파드를 예약하는 방법
+
+### 데몬셋 컨트롤러로 예약(1.12 이후로 기본값으로 사용 안함)
+
+일반적으로 쿠버네티스 스케줄러에 의해 파드가 실행되는 머신이 선택된다. 그러나 데몬셋 
+컨트롤러에 의해 생성된 파드는 이미 머신이 선택되어 있다(`.spec.nodeName` 은 
+파드가 생성될 때 명시되고, 스케줄러에 의해 무시된다). 그러므로 다음과 같다.
+
+ - 데몬셋 컨트롤러는 노드의 [`unschedulable`](/ko/docs/concepts/architecture/nodes/#수동-노드-관리) 
+   필드를 존중하지 않는다.
+ - 데몬셋 컨트롤러는 스케줄러가 시작되지 않은 경우에도 파드를 만들 수 있아 클러스터 부트스트랩에 
+   도움이 된다.
+
+
+### 기본 스케줄러로 예약(1.12 이후로 기본값으로 사용 가능)
+
+{{< feature-state state="beta" for-kubernetes-version="1.12" >}}
+
+데몬셋은 모든 자격이 되는 노드에서 파드 사본이 실행하도록 보장한다. 일반적으로 
+쿠버네티스 스케줄러에 의해 파드가 실행되는 노드가 선택된다. 그러나 
+데몬셋 파드는 데몬셋 컨트롤러에 의해 생성되고 예약된다.
+이에 대한 이슈를 소개한다.
+
+ * 파드 동작의 불일치: 예약되어 대기중인 일반 파드는 `Pending` 상태로 생성된다.
+   그러나 데몬셋 파드는 `Pending` 상태로 생성되지 않는다.
+   이것은 사용자에게 혼란을 준다.
+ * [파드 선점](/docs/concepts/configuration/pod-priority-preemption/)
+   은 기본 스케줄러에서 처리한다. 선점이 활성화되면 데몬셋 컨트롤러는 
+   파드 우선순위와 선점을 고려하지 않고 예약한다.
+
+`ScheduleDaemonSetPods` 로 데몬셋 파드에 `.spec.nodeName` 용어 대신 
+`NodeAffinity` 용어를 추가해서 데몬셋 컨트롤러 대신 기본 
+스케줄러를 사용해서 데몬셋을 예약할 수 있다. 이후에 기본 
+스케줄러를 사용해서 대상 호스트에 파드를 바인딩 한다. 만약 데몬셋 파드에 
+이미 노드 선호도가 존재하면 교체 한다. 데몬셋 컨트롤러는 
+데몬셋 파드를 만들거나 수정할때만 이런 작업을 수행하며, 
+데몬셋의 `spec.template` 은 변경되지 않는다.
+
+```yaml
+nodeAffinity:
+  requiredDuringSchedulingIgnoredDuringExecution:
+    nodeSelectorTerms:
+    - matchFields:
+      - key: metadata.name
+        operator: In
+        values:
+        - target-host-name
+```
+
+또한, 데몬셋 파드에 `node.kubernetes.io/unschedulable:NoSchedule` 이 톨러레이션으로 
+자동으로 추가된다. 기본 스케줄러는 데몬셋 파드를 
+스케줄링시 `unschedulable` 노드를 무시한다.
+
+
+### 테인트와 톨러레이션(Taints and Tolerations)
+
+데몬 파드는 
+[테인트와 톨러레이션](/docs/concepts/configuration/taint-and-toleration)을 존중하지만, 
+관련 기능기능에 따라 자동적으로 데몬셋 파드에 다음과 같이 
+톨러레이션을 추가한다.
+
+| Toleration Key                           | Effect     | Version | Description                                                  |
```

Comment:
These headings are not localized.

Reply:
Translation has been added for this part.